src/components/EnterpriseList/index.jsx

```diff
@@ -3,178 +3,87 @@ import PropTypes from 'prop-types';
-        },
-      ],
-      enterprises: enterprises && enterprises.results,
-      pageCount: enterprises && enterprises.num_pages,
       searchQuery: queryParams.search || '',
       searchSubmitted: !!queryParams.search,
     };
-
-    this.formatEnterpriseData = this.formatEnterpriseData.bind(this);
   }
 
   componentDidMount() {
-    const options = qs.parse(this.props.location.search);
-    this.clearPortalConfiguration();
-    this.getEnterpriseList(formatTableOptions(options));
-    this.props.getLocalUser();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { enterprises, searchQuery } = this.props;
-
-    if (enterprises !== prevProps.enterprises) {
-      this.setState({ // eslint-disable-line react/no-did-update-set-state
-        enterprises: enterprises && enterprises.results,
-        pageCount: enterprises && enterprises.num_pages,
-      });
-    }
-    if (searchQuery !== prevProps.searchQuery) {
-      const options = this.formatQueryParams({ search: searchQuery });
-      this.props.history.push(`?${qs.stringify(options)}`);
-      this.getEnterpriseList(options);
-    }
-  }
-
-  getEnterpriseList(options) {
-    this.props.getEnterpriseList(options);
-  }
-
-  clearPortalConfiguration() {
     this.props.clearPortalConfiguration();
-  }
-
-  formatQueryParams(options) {
-    return omitBy({
-      search: options.search,
-    }, isEmpty);
-  }
-
-  formatEnterpriseData(enterprises) {
-    if (!enterprises) {
-      return null;
-    } else if (!enterprises.length) {
-      return [];
-    }
-
-    return enterprises.map(enterprise => ({
-      link: <Link to={`/${enterprise.slug}/admin`}>{enterprise.name}</Link>,
-      name: enterprise.name,
-      slug: enterprise.slug,
-      uuid: enterprise.uuid,
-    }));
+    this.props.getLocalUser();
   }
 
   handleSearch(query) {
-    this.props.setSearchQuery(query);
     this.setState({
       searchQuery: query,
       searchSubmitted: true,
     });
-  }
-
-  handleClear() {
-    this.props.setSearchQuery('');
-    this.setState({
-      searchQuery: '',
+    this.props.searchEnterpriseList({
+      search: query || undefined,
     });
   }
 
-  shouldRenderRedirectToEnterpriseAdminPage() {
-    const { loading, error } = this.props;
-    const { enterprises, pageCount, searchSubmitted } = this.state;
-    return (
-      !loading && !error && enterprises && enterprises.length === 1 &&
-        pageCount === 1 && !searchSubmitted
-    );
-  }
+  formatEnterpriseData = enterprises => enterprises.map(enterprise => ({
+    link: <Link to={`/${enterprise.slug}/admin`}>{enterprise.name}</Link>,
+    name: enterprise.name,
+    slug: enterprise.slug,
+    uuid: enterprise.uuid,
+  }));
 
-  renderEmptyEnterpriseListMessage() {
-    const { searchQuery } = this.props;
-    let emptyMessage = 'You do not have permission to view any enterprise customer data.';
-    if (searchQuery) {
-      emptyMessage = 'There are no enterprise customers that match your search.';
```

Comment:
Are we ok with not having custom empty state messages within `TableComponent`? With the refactor, it would always say "There are no results." In some scenarios, we may want to change that message.

Reply:
We could always add in a props for overriding if we need to... I figured `There are no results.` was generic enough to work in both use-cases we have now (the enterprise list and the data tables) but if we want I could add the `emptyMessage` props now